src/allmydata/storage/server.py

```diff
@@ -331,7 +331,31 @@ def _iter_share_files(self, storage_index):
             yield sf
 
     def remote_add_lease(self, storage_index, renew_secret, cancel_secret,
-                         owner_num=1):
+                         owner_num=1, policy=None):
```

Comment:
Corresponding to my comment above, then, instead of adding `policy` here, the plugin would provide a new object with an `remote_add_lease` method and a private reference to an instance of *this* class.  The policy is just "self".

This also gives an example of why the plugin may want to be able to alter args ... though it's tenuous at best: perhaps the plugin can turn credentials into an "owner_num" and provide a different value to the underlying system (but I don't think owner_num is used for anything so it's pretty academic).

Reply:
Just to clarify in case it's not obvious, the `policy=` kwarg there is passing along any data the *client* of the plugin needs to communicate to the storage server (i.e. just `token=` for the privacy-pass based plugin). (see below for some words about mutating args)